tests/providers/test_braket_quantum_task.py

```diff
@@ -47,6 +54,50 @@ def test_result(self):
         self.assertEqual(task.result().results[0].shots, 3)
         self.assertEqual(task.result().get_memory(), ["10", "10", "01"])
 
+    @patch(
+        "qiskit_braket_provider.providers.braket_quantum_task.AwsQuantumTaskBatch._retrieve_results"
+    )
+    def test_task_result_is_none(self, mock_retrieve_results):
+        """Tests result when result is None"""
+        mock_retrieve_results.return_value = [None, None]
+
+        task = BraketQuantumTask(
+            backend=BraketLocalBackend(name="default"),
+            task_id="AwesomeId",
+            tasks=[MOCK_LOCAL_QUANTUM_TASK, MOCK_LOCAL_QUANTUM_TASK],
+            shots=10,
+        )
+        task.result()
```

Comment:
Since the test is called `test_task_result_is_none`, I think assert statements about result being None should be added here